src/store/slices/node/websocketMiddleware.ts

```diff
@@ -0,0 +1,70 @@
+import { api, utils } from '@hoprnet/hopr-sdk';
+import { Middleware, PayloadAction } from '@reduxjs/toolkit';
+import { initialState } from '../auth/initialState';
+import { nodeActions } from './index';
+
+const {
+  messageReceived,
+  initializeWebsocket,
+  closeWebsocket,
+  updateWebsocketStatus,
+} = nodeActions;
+
+const { getWsUrl } = api;
+const { WebsocketHelper } = utils;
+
+type LocalRootState = {
+  auth: typeof initialState;
+};
+
+const websocketMiddleware: Middleware<{}, LocalRootState> = ({
+  dispatch,
+  getState,
+}) => {
+  let socket: typeof WebsocketHelper.prototype | null = null;
+
+  return (next) => (action: PayloadAction) => {
+    if (action.type === initializeWebsocket.type) {
+      // start websocket connection
+      const { apiEndpoint, apiToken } = getState().auth.loginData;
+      if (apiEndpoint && apiToken) {
+        const wsUrl = getWsUrl(
+          apiEndpoint,
+          '/api/v2/messages/websocket/',
+          apiToken
+        );
+        try {
+          socket = new WebsocketHelper({
+            url: wsUrl,
```

Comment:
We should update the WebsocketHelper to comly with the apiEndpoint and apiToken standard we have in all the other calls, and then use it here